tests/test_dataset_consistency.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import json
+from collections import Counter
+from pathlib import Path
+from typing import Any
+
+import pytest
+import yaml
+
+# Search both locations
+BASE_DIRS = [Path("src/data"), Path("src/mapped_data")]
+
+
+def _pick_existing_pairs() -> list[tuple[Path, str, str]]:
+    candidates = [
+        # in-place
+        ("nlu.yaml", "samples.train.jsonl"),
+        ("validation.yaml", "samples.validation.jsonl"),
+        ("test.yaml", "samples.test.jsonl"),
+        # same-dir suffix
+        ("nlu.mapped.yaml", "samples.train.mapped.jsonl"),
+        ("validation.mapped.yaml", "samples.validation.mapped.jsonl"),
+        ("test.mapped.yaml", "samples.test.mapped.jsonl"),
+        # out-dir (no suffix, when using --out-dir)
+        ("nlu.yaml", "samples.train.jsonl"),
+        ("validation.yaml", "samples.validation.jsonl"),
+        ("test.yaml", "samples.test.jsonl"),
+    ]
+    pairs: list[tuple[Path, str, str]] = []
+    for base in BASE_DIRS:
+        for y, j in candidates:
+            ypath, jpath = base / y, base / j
+            if ypath.exists() and jpath.exists():
+                pairs.append((base, ypath.name, jpath.name))
+    return pairs
+
+
+def _flatten_yaml(yaml_path: Path) -> list[dict[str, Any]]:
+    doc: dict[str, Any] = yaml.safe_load(yaml_path.read_text(encoding="utf-8")) or {}
+    rows: list[dict[str, Any]] = []
+    for item in doc.get("nlu") or []:
+        if not isinstance(item, dict):
+            continue
+        parent: str = str(item.get("intent") or "OTHER")
+        sub: str | None = (
+            item.get("subintent") if isinstance(item.get("subintent"), str) else None
+        )
+        examples_block: str = str(item.get("examples") or "")
+        examples = [
+            ln[2:].strip()
+            for ln in examples_block.splitlines()
+            if ln.strip().startswith("- ")
+        ]
+        for t in examples:
+            rec: dict[str, Any] = {"text": t, "label": parent}
+            if parent == "FEEDBACK" and sub in {"COMPLIMENT", "COMPLAINT"}:
+                rec["feedback_subtype"] = sub
+            if parent == "SENSITIVE_EXIT" and sub in {"BABY_LOSS", "OPTOUT"}:
+                rec["sensitive_exit_subtype"] = sub
+            if parent == "OTHER" and sub in {
+                "ACCOUNT_UPDATE",
+                "INFORMATION_QUERY",
+                "CONFIRMATION",
+            }:
+                rec["other_subtype"] = sub
+            rows.append(rec)
+    rows.sort(key=lambda r: (str(r["label"]), str(r["text"]).casefold()))
+    return rows
+
+
+def _load_jsonl(jsonl_path: Path) -> list[dict[str, Any]]:
+    return [
+        json.loads(line)
+        for line in jsonl_path.read_text(encoding="utf-8").splitlines()
+        if line.strip()
+    ]
+
+
+def test_yaml_and_jsonl_counts_and_labels_match() -> None:
+    pairs = _pick_existing_pairs()
```

Comment:
For some of these tests a docstring might help others understand what we are testing here.